asistente_ladm_col/lib/geometry.py

```diff
@@ -1112,3 +1112,20 @@ def get_dangle_ids(self, boundary_layer):
     def create_spatial_index(layer):
         if layer.hasSpatialIndex() != QgsFeatureSource.SpatialIndexPresent:
             processing.run("native:createspatialindex", {'INPUT': layer})
+
+    @staticmethod
+    def z_coordinate_is_nan(layer):
+        """
+        Check if layer have any z-coordinate with nan value
+        :param layer: QgsVectorLayer
+        :return: True if any z-coordinate with nan value, False in other case
+        """
+
+        if layer.isSpatial():
+            nodes_layer = layer
+            if layer.geometryType() != QgsWkbTypes.PointGeometry or QgsWkbTypes.isMultiType(layer.wkbType()):
```

Comment:
Again undocumented key parts of the code...

What's the rationale of this?

`layer.geometryType() != QgsWkbTypes.PointGeometry or QgsWkbTypes.isMultiType(layer.wkbType())`

Reply:
When the layer is not of type point or is multigeometry, the vertices are extracted because the function z($geometry) only works for geometry of type point.

Reply:
So `z_coordinate_is_nan()` won't work for point layers nor for single type layers... Very inflexible.

Hints:

 + If its a point layer just don't extract vertices...
 + If it's a multitype layer you just need to convert to a single type layer before...

I don't see any reason why we cannot support points or single type layers in `z_coordinate_is_nan()`.